src/M1/M1_implicitstep.F90

```diff
@@ -1168,10 +1168,9 @@ subroutine M1_implicitstep(dts,implicit_factor)
                        
                        !second time in here, make flux implicit
                        if (trouble_brewing) then
-                          sourceG(problem_zone,3) = q_M1_old(k,i,problem_zone,1) + &
-                               D_M1(k,i,j,1)
```

Comment:
This is what you mention you changed for some reason.  I guess this is more stable, and I think I'm fine with this since it is in the 'changed twice' regime.